doc/samples/sample_notify_fifo.sh

```diff
@@ -60,6 +62,17 @@ if [[ ! -p $FIFO ]]; then
 	fi
 fi
 
+# wait for a previous instance of this script to finish its job
+while [ -f "$PID_FILE" ] && \
+	ls -l /proc/$(cat "$PID_FILE")/fd | grep -qw "$(readlink -f "$PID_FILE")"
+do
+	sleep 0.05
+done
```

Comment:
Should there be a time limit (or loop count limit) here so that the script does not wait indefinitely for a previous instance to terminate? Perhaps if the previous instance of the script hasn't terminated after a couple of seconds it should be killed.

Reply:
One of my collegues suggested to me to replace the loop by a flock.
Basically, flock takes a exclusive lock on the FD until FD is removed with `exec {FD}>&-` or at exit.
```
exec {FD}>>"$PID_FILE"
if ! flock -e $FD --timeout 10; then
	if ls -l /proc/$(cat "$PID_FILE")/fd | grep -qw "$(readlink -f "$PID_FILE")"
	then
		kill $(cat "$PID_FILE")
	else
		exit 1
	fi
	flock -e $FD --timeout 10 || exit 1
fi
echo $$ >"$PID_FILE"
```
We can use flock without the timeout. Just:
`flock -e $FD`

However it would wait indefinitely if the previous script does not remove the the FD or exit.

Reply:
I note that the man page for flock(1) states that the syntax is `flock [options] number`, and so to comply with that I think it should be `flock -e --timeout 10 $FD`.

I think the timeout should be more like 1 second or even less rather than 10 seconds; the previous instance of the script should be terminating within a few milli-seconds of the script receiving SIGTERM. Perhaps have `TIMEOUT=0.3` near the beginning of the script and use `flock -e --timeout $TIMEOUT $FD` in the two places where flock is called. Users who are then using the sample script can modify $TIMEOUT as fits their needs.